dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkflowDefinitionServiceImpl.java

```diff
@@ -2122,6 +2124,19 @@ protected void doBatchOperateWorkflowDefinition(User loginUser,
                         long taskCode = CodeGenerateUtils.genCode();
                         taskCodeMap.put(taskDefinitionLog.getCode(), taskCode);
                         taskDefinitionLog.setCode(taskCode);
+                        if (TaskTypeUtils.isSwitchTask(taskDefinitionLog.getTaskType())){
+                            final String taskParams = taskDefinitionLog.getTaskParams();
+                            final SwitchParameters switchParameters = JSONUtils.parseObject(taskParams, SwitchParameters.class);
+                            if (switchParameters == null) {
+                                throw new IllegalArgumentException("Switch task params: " + taskParams + " is invalid.");
+                            }
+                            List<SwitchResultVo> dependTaskList = switchParameters.getSwitchResult().getDependTaskList();
+                            dependTaskList.forEach(switchResultVo -> {
+                                switchResultVo.setNextNode(taskCodeMap.get(switchResultVo.getNextNode()));
```

Comment:
We also need to change the default branch in `SwitchResult`.

Reply:
> 我们还需要更改默认分支`SwitchResult`。

done